src/metax_api/services/file_service.py

```diff
@@ -890,17 +890,25 @@ def _create_single(cls, common_info, initial_data, serializer_class, **kwargs):
         return res
 
     @classmethod
-    def verify_allowed_projects(cls, allowed_projects, file_identifiers=[]):
-        if file_identifiers:
+    def check_allowed_projects(cls, request):
+        allowed_projects = CommonService.get_list_query_param(request, 'allowed_projects')
+
+        if allowed_projects is not None:
+            if not isinstance(request.data, list):
+                raise Http400({ 'detail': [ 'request.data is not a list' ] })
```

Comment:
More general error message: "request message body must be a list of json objects" etc